components/nimbus/src/evaluator.rs

```diff
@@ -332,47 +365,109 @@ mod tests {
         custom_targeting_attributes.insert("is_first_run".into(), "true".into());
         custom_targeting_attributes.insert("ios_version".into(), "8.8".into());
         // A matching context that includes the appropriate specific context
-        let ctx = AppContext {
-            app_name: "nimbus_test".to_string(),
-            app_id: "1010".to_string(),
-            channel: "test".to_string(),
-            app_version: Some("4.4".to_string()),
-            app_build: Some("1234".to_string()),
-            architecture: Some("x86_64".to_string()),
-            device_manufacturer: Some("Samsung".to_string()),
-            device_model: Some("Galaxy S10".to_string()),
-            locale: Some("en-US".to_string()),
-            os: Some("Android".to_string()),
-            os_version: Some("10".to_string()),
-            android_sdk_version: Some("29".to_string()),
-            debug_tag: None,
-            custom_targeting_attributes: Some(custom_targeting_attributes),
+        let targeting_attributes = TargetingAttributes {
+            app_context: AppContext {
+                app_name: "nimbus_test".to_string(),
+                app_id: "1010".to_string(),
+                channel: "test".to_string(),
+                app_version: Some("4.4".to_string()),
+                app_build: Some("1234".to_string()),
+                architecture: Some("x86_64".to_string()),
+                device_manufacturer: Some("Samsung".to_string()),
+                device_model: Some("Galaxy S10".to_string()),
+                locale: Some("en-US".to_string()),
+                os: Some("Android".to_string()),
+                os_version: Some("10".to_string()),
+                android_sdk_version: Some("29".to_string()),
+                debug_tag: None,
+                custom_targeting_attributes: Some(custom_targeting_attributes),
+            },
```

Comment:
Nit: there's lots of these blocks around. Perhaps we could put them in a single method?

Reply:
I made it a lot simpler using `From` for now, I'll file a follow up to clean up the tests so the `AppContext` is constructed in a helper